stock_picking_volume/models/stock_move.py

```diff
@@ -23,27 +23,10 @@ def _compute_volume(self):
             qty = move.product_uom_qty
             if move.state in ("partially_available", "assigned"):
                 qty = move.reserved_availability
-            new_volume = move._get_volume_for_qty(qty=qty)
+            new_volume = move.product_id._get_volume_for_qty(qty, move.product_uom)
```

Comment:
@mt-software-de We can move the compute method to the product.product model but we could still keep the method `_get_volume_for_qty` defined on the stock.move forwarding the call to the product.  In this way we keep an extension point if we want to adapt the way the weight is computed according to information on the stock.move and we don't break the current api.

Reply:
The compute method is still defined on stock.move. 
We can keep _get_volume_for_qty as it is on the stock.move but i can't see any advantage of that. 
Also, the development status beta would allow changing the current API. https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/oca_module_lifecycle_development_status.rst#id2 

If we would keep it, i would also change it to something like this.
`
    @api.depends("product_id", "product_uom_qty", "state", "move_line_ids.product_qty")
    def _compute_volume(self):
        for move in self:
            new_volume = move._get_volume()
            if move.volume != new_volume:
                move.volume = new_volume
                    
    def _get_volume(self):
        self.ensure_one()
        qty = self.product_uom_qty
        if self.state in ("partially_available", "assigned"):
            qty = self.reserved_availability
        return self.product_id._get_volume_for_qty(qty, self.product_uom)
`

Reply:
Ok then i will leave it as it is